step/wait_for_signal.go

```diff
@@ -45,12 +58,16 @@ func WaitForSignal[T any](ctx context.Context, stepID string, opts WaitForSignal
 	mgr := preflight(ctx)
 
 	args := map[string]any{
-		"signal":  opts.Signal,
-		"timeout": str2duration.String(opts.Timeout),
+		"signal":     opts.Signal,
+		"timeout":    str2duration.String(opts.Timeout),
+		"onConflict": SignalConflictFail,
 	}
 	if opts.Name == "" {
 		opts.Name = stepID
 	}
+	if opts.OnConflict != "" {
+		args["onConflict"] = opts.OnConflict
```

Comment:
The option the Executor looks for is just `"conflict"`
```suggestion
		args["conflict"] = opts.OnConflict
```